From 27be25a99e91bd26ee4eefd78ddda0b333f0c30d Mon Sep 17 00:00:00 2001
From: Yousong Zhou <zhouyousong@yunionyun.com>
Date: Mon, 22 Oct 2018 09:50:46 +0000
Subject: [PATCH] Fix "_zsh_autosuggest_modify:local:27: not valid in this
 context: zsh-autosuggestions/"

This can happen in the following key sequence

	()<Left><Space>
	cd auto<Tab>
---
 zsh-autosuggestions.zsh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zsh-autosuggestions.zsh b/zsh-autosuggestions.zsh
index 1c3eab5..04b5775 100644
--- a/zsh-autosuggestions.zsh
+++ b/zsh-autosuggestions.zsh
@@ -342,7 +342,7 @@ _zsh_autosuggest_modify() {
 
 	# Optimize if manually typing in the suggestion
 	if (( $#BUFFER > $#orig_buffer )); then
-		local added=${BUFFER#$orig_buffer}
+		local added="${BUFFER#$orig_buffer}"
 
 		# If the string added matches the beginning of the postdisplay
 		if [[ "$added" = "${orig_postdisplay:0:$#added}" ]]; then
